### PR TITLE
Added a badgeColor to the ListRecords Tab Component

### PR DESCRIPTION
**Pull request by schmeits**
- [V] Changes have been thoroughly tested to not break existing functionality.
- [V] New functionality has been documented or existing documentation has been updated to reflect changes.
- [V] Visual changes are explained in the PR description using a screenshot/recording of before and after.

### CODE REVIEW

File: packages/panels/src/Resources/Pages/ListRecords/Tab.php [hunk 1]
```diff
@@ -15,6 +15,8 @@ class Tab extends Component
 
     protected string | Closure | null $badge = null;
 
+    protected string | Closure | null $badgeColor = null;
```
Comment (zepfietje) [inline packages/panels/src/Resources/Pages/ListRecords/Tab.php]: Should accept `string | array | Closure | null`.
Could you update and mark as ready for review when done, @schmeits?

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/panels/src/Resources/Pages/ListRecords/Tab.php (hunk 1 above)
Comment (schmeits) [reply packages/panels/src/Resources/Pages/ListRecords/Tab.php]: Yes changed it... thanks for the tip.. completely forgot the array color option

File: packages/panels/src/Resources/Pages/ListRecords/Tab.php [hunk 2]
```diff
@@ -68,6 +83,14 @@ public function getBadge(): ?string
         return $this->evaluate($this->badge);
     }
 
+    /**
+     * @return string | array{50: string, 100: string, 200: string, 300: string, 400: string, 500: string, 600: string, 700: string, 800: string, 900: string, 950: string} | null
+     */
+    public function getBadgeColor(): string | array | Closure | null
```
Comment (zepfietje) [inline packages/panels/src/Resources/Pages/ListRecords/Tab.php]: This shouldn't ever return a closure, right?

Comment (schmeits) [reply packages/panels/src/Resources/Pages/ListRecords/Tab.php]: You're completely right... forgot this one... sorry :)

Comment (schmeits) [reply packages/panels/src/Resources/Pages/ListRecords/Tab.php]: Isn't it nicer to extract the code to a HasBadgeColor concern/trait?

Comment (zepfietje) [reply packages/panels/src/Resources/Pages/ListRecords/Tab.php]: Potentially. Maybe if we use it in other places too?

Comment (schmeits) [reply packages/panels/src/Resources/Pages/ListRecords/Tab.php]: or maybe shoud have used the HasBadge in the Tab component?

Comment (schmeits) [reply packages/panels/src/Resources/Pages/ListRecords/Tab.php]: works as well... think that's the best option for now... no more duplication

Comment (zepfietje) [reply packages/panels/src/Resources/Pages/ListRecords/Tab.php]: Sounds good, though that trait is from the `actions` package.
Could you move it into `support`?

Comment (schmeits) [reply packages/panels/src/Resources/Pages/ListRecords/Tab.php]: Yes did it and updated all the other references

File: packages/panels/docs/03-resources/02-listing-records.md [hunk 3]
```diff
@@ -76,6 +76,26 @@ Tab::make()
 
 As in the example above, this could be quite useful for showing the number of records that pass that filter.
 
+### Customizing the color of the badges in filter tabs
+
+You can change the color of the badges in the tabs by passing a string into the `badgeColor()` method of the tab:
```
Comment (zepfietje) [inline packages/panels/docs/03-resources/02-listing-records.md]:
```suggestion
You can change the color of a badge using the `badgeColor()` method:
```

File: packages/panels/docs/03-resources/02-listing-records.md [hunk 4]
```diff
@@ -76,6 +76,26 @@ Tab::make()
 
 As in the example above, this could be quite useful for showing the number of records that pass that filter.
 
+### Customizing the color of the badges in filter tabs
```
Comment (zepfietje) [inline packages/panels/docs/03-resources/02-listing-records.md]:
```suggestion
```

File: packages/panels/docs/03-resources/02-listing-records.md [hunk 5]
```diff
@@ -76,6 +76,25 @@ Tab::make()
 
 As in the example above, this could be quite useful for showing the number of records that pass that filter.
 
+
```
Comment (zepfietje) [inline packages/panels/docs/03-resources/02-listing-records.md]:
```suggestion
```

File: packages/panels/docs/03-resources/02-listing-records.md [hunk 6]
```diff
@@ -76,6 +76,24 @@ Tab::make()
 
 As in the example above, this could be quite useful for showing the number of records that pass that filter.
 
+You can change the color of a badge using the `badgeColor()` method:
```
Comment (zepfietje) [inline packages/panels/docs/03-resources/02-listing-records.md]:
```suggestion
The color of a badge may be changed using the `badgeColor()` method:
```

File: packages/panels/docs/03-resources/02-listing-records.md [hunk 7]
```diff
@@ -76,6 +76,24 @@ Tab::make()
 
 As in the example above, this could be quite useful for showing the number of records that pass that filter.
 
+The color of a badge may be changed using the `badgeColor()` method:
+
+```php
+use Filament\Resources\Pages\ListRecords\Tab;
+
+Tab::make()
+    ->badgeColor('warning')
```
Comment (zepfietje) [inline packages/panels/docs/03-resources/02-listing-records.md]:
```suggestion
    ->badge(Customer::query()->where('active', true)
    ->badgeColor('success')
```

File: packages/panels/docs/03-resources/02-listing-records.md [hunk 8]
```diff
@@ -76,6 +76,25 @@ Tab::make()
...
+
+You can also pass in a closure to change the color depending on the count or name of the tab.
+
+```php
+use Filament\Resources\Pages\ListRecords\Tab;
+
+Tab::make()
+    ->badgeColor(fn() => (Customer::query()->where('active', true)->count() > 0 ? 'primary' : 'danger'))
+```
+
```
Comment (zepfietje) [inline packages/panels/docs/03-resources/02-listing-records.md]:
```suggestion

```

File: packages/panels/docs/03-resources/02-listing-records.md [hunk 9]
```diff
@@ -76,6 +76,15 @@ Tab::make()
...
+The color of a badge may be changed using the `badgeColor()` method:
+
+```php
+use Filament\Resources\Pages\ListRecords\Tab;
+
+Tab::make()
+    ->badge(Customer::query()->where('active', true)
+    ->badgeColor('success')
+```
 ### Customizing the default tab
```
Comment (zepfietje) [inline packages/panels/docs/03-resources/02-listing-records.md:89]:
```suggestion

### Customizing the default tab
```

File: packages/panels/src/Resources/Pages/ListRecords/Tab.php [hunk 10]
```diff
@@ -4,17 +4,16 @@
...
 use Closure;
 use Filament\Support\Components\Component;
+use Filament\Support\Concerns\HasBadge;
 use Filament\Support\Concerns\HasIcon;
 use Illuminate\Database\Eloquent\Builder;
 
 class Tab extends Component
 {
-    use HasIcon;
+    use HasIcon, HasBadge;
```
Comment (zepfietje) [inline packages/panels/src/Resources/Pages/ListRecords/Tab.php]:
```suggestion
    use HasBadge;
    use HasIcon;
```

**Review by zepfietje — APPROVED** [review]
LGTM

**Pull request merged**